src/files/examples/pyfingerprint_examples.py

```diff
@@ -0,0 +1,258 @@
+	## Tries to search the finger and calculate hash
+	try:
+		print('Waiting for finger...')
+
+		## Wait that finger is read
+		while ( f.readImage() == False ):
+			pass
+
+		## Converts read image to characteristics and stores it in charbuffer 1
+		logging.debug("Image read converting it")
+		f.convertImage(0x01)
+
+		## Searchs template
+		result = f.searchTemplate()
+
+		position_number = result[0]
+		accuracyScore = result[1]
+
+		if ( position_number == -1 ):
+			print('No match found!')
+			exit(0)
+		else:
+			print('Found template at position #' + str(position_number))
+			print('The accuracy score is: ' + str(accuracyScore))
+
+		## OPTIONAL stuff
+		##
+
+		## Loads the found template to charbuffer 1
+		f.loadTemplate(position_number, 0x01)
+
+		## Downloads the characteristics of template loaded in charbuffer 1
+		characteristics = str(f.downloadCharacteristics(0x01)).encode('utf-8')
+		logging.debug("Characteristics raw:%s" % (characteristics))
+
+		## Hashes characteristics of template
+		print('SHA-2 hash of template: ' + hashlib.sha256(characteristics).hexdigest())
+
+	except Exception as e:
+		logging.exception('Could not display index table')
+		raise	
+
+def enroll():
+	"""Asks for fingerprint and adds it to database"""
+	try:
+		print('Waiting for finger...')
+
+		## Wait that finger is read
+		while ( f.readImage() == False ):
+			pass
+
+		## Converts read image to characteristics and stores it in charbuffer 1
+		logging.debug("Image read, converting it")
+		f.convertImage(0x01)
+
+		## Checks if finger is already enrolled
+		result = f.searchTemplate()
+		position_number = result[0]
+
+		if ( position_number >= 0 ):
+			print('Template already exists at position #' + str(position_number))
+			exit(0)
+
+		print('Remove finger...')
+		time.sleep(2)
+
+		print('Waiting for same finger again...')
+
+		## Wait that finger is read again
+		while ( f.readImage() == False ):
+			pass
+
+		## Converts read image to characteristics and stores it in charbuffer 2
+		f.convertImage(0x02)
+
+		## Compares the charbuffers
+		if ( f.compareCharacteristics() == 0 ):
+			raise Exception('Fingers do not match')
+
+		## Creates a template
+		f.createTemplate()
+
+		## Saves template at new position number
+		position_number = f.storeTemplate()
+		print('Finger enrolled successfully!')
+		print('New template position #' + str(position_number))
+
+	except Exception as e:
+		logging.exception('Could not enroll your fingerprint')
+		raise
+
+def display_index_table():
+	"""Displays the index tables of stored fingerprints"""
+	try:
+		logging.info('Currently used templates: %s/%s' % (f.getTemplateCount(), f.getStorageCapacity()))
+		for page in range(0, 4):
+			tableIndex = f.getTemplateIndex(page)
+			s = "";
+			for i in range(0, len(tableIndex)):
+				s = s + "p.%s:index:%s:%s|" % (page, i, tableIndex[i])
+			print(s)
+	except Exception as e:
+		logging.exception('Could not display index table')
+		raise
+
+def print_random_number():
```

Comment:
I suggest `generate_random_number()`

Reply:
Think generate would affect some variable or maybe return it as it would be a getXXX would print_generated_random_number() be ok 4 you?